canal/local.go

```diff
@@ -0,0 +1,108 @@
+package canal
+
+import (
+	"context"
+
+	"github.com/go-mysql-org/go-mysql/mysql"
+	"github.com/go-mysql-org/go-mysql/replication"
+	"github.com/pingcap/errors"
+)
+
+// BinlogFileDownload download the binlog file from cloud computing platform (etc. aliyun)
+type BinlogFileDownload func(mysql.Position) (localBinFilePath string, err error)
+
+// WithLocalBinlogDownload registers the local bin file download,
+// that allows download the flushed binlog file to local (etc. aliyun)
+func (c *Canal) WithLocalBinlogDownload(d BinlogFileDownload) {
+	c.binFileDownload = d
+}
+
+func (c *Canal) adaptLocalBinFileStreamer(syncMasterStreamer *replication.BinlogStreamer, err error) (*LocalBinFileAdapterStreamer, error) {
```

Comment:
how about use `remoteBinlogStreamer` rather than `syncMasterStreamer`? local vs remote

Reply:
I only change the function args name, is that ok?